code/game/mecha/mecha.dm

```diff
@@ -1346,13 +1346,19 @@
 	lock_dir = !lock_dir
 
 /obj/mecha/MouseDropFrom(over_object, src_location, var/turf/over_location, src_control, over_control, params)
-	if(usr != src.occupant || usr.incapacitated())
+	if(!Adjacent(over_location))
+		return
+	if(!istype(over_location) || over_location.density)
 		return
 	if(istype(occupant, /mob/living/carbon/brain))
 		return
-	if(!istype(over_location) || over_location.density)
+	if(usr.incapacitated())
 		return
-	if(!Adjacent(over_location))
+	if(usr != occupant)
+		visible_message("<span class='notice'>[usr] start pulling [occupant.name] out of \the [src].</span>")
+		if(do_after(usr, src, 4 SECONDS))
```

Comment:
If you stun the mech pilot with a banger you've got enough time to pull them out.
Intended ?

Reply:
huh, i just copied the code from my spacepod pr.

Im gonna raise it to 10 seconds, since this is intended to be used if someone is knocked out inside a mech